app/docs/0.10.x/proxy.md

```diff
@@ -0,0 +1,759 @@
+conditions. The first request's URI is not a match for any of the configured 
+`uris`, and it is so for the second request's HTTP method, or the third's Host
+header.
+
+Now that we grasp how each of those three properties play together, let's go
+into more details and explore each one of them individually.
+
+[Back to TOC](#table-of-contents)
+
+#### Request Host header
+
+It shouldn't come as a surprise that routing a request based on its Host header
+is the most straightforward way to proxy traffic through Kong, as this is the
+intended usage of the HTTP Host header itself. Hence, Kong certainly makes it 
+easy to do so via the `hosts` field of the API entity.
+
+This field accepts multiple values, which must be comma-separated when 
+specifying them via the Admin API:
+
+```bash
+$ curl -i -X POST http://localhost:8001/apis/ \
+    -d 'name=my-api' \
+    -d 'upstream_url=http://my-api.com' \
+    -d 'hosts=my-api.com,example.com,service.com'
+HTTP/1.1 201 Created
+...
+```
+
+To satisfy the `hosts` condition of this API, any incoming request from a client
+must now have its Host header set to one of:
+
+```
+Host: my-api.com
+```
+
+or:
+
+```
+Host: example.com
+```
+
+or:
+
+```
+Host: service.com
+```
+
+[Back to TOC](#table-of-contents)
+
+##### Using wildcard hostnames
+
+To provide flexibility, Kong allows you to specify hostnames with wildcards in
+the `hosts` field. Such hostnames allow any matching Host header to satisfy the
+condition, and match a given API. 
+
+Such hostnames **must** contain **only one** asterisk at the leftmost **or** 
+rightmost label of the domain. Examples:
+
+- `*.example.org` would allow Host values such as `a.example.com` and
+`x.y.example.com` to match.
+- `example.*` would allow Host values such as `example.com` and `example.org` to
+match.
+
+A complete example would look like this:
+
+```json
+{
+    "name": "my-api",
+    "upstream_url": "http://my-api.com",
+    "hosts": ["*.example.com", "service.com"]
+}
+```
+
+And allow the following requests to match this API:
+
+```http
+GET / HTTP/1.1
+Host: an.example.com
+```
+
+```http
+GET / HTTP/1.1
+Host: service.com
+```
+
+[Back to TOC](#table-of-contents)
+
+##### The `preserve_host` property
+
+By default when proxying, Kong's behavior is to set the upstream request's Host
+header to the hostname of the API's `upstream_url` property. The `preserve_host`
+field accepts a boolean flag instructing Kong not to do so.
+
+Example:
+
+```json
+{
+    "name": "my-api",
+    "upstream_url": "http://my-api.com",
+    "hosts": ["service.com"],
+}
+```
+
+A possible request from a client to Kong could be:
+
+```http
+GET / HTTP/1.1
+Host: service.com
+```
+
+By default, Kong would send the following request to your upstream service:
+
+```http
+GET / HTTP/1.1
+Host: my-api.com
+```
+
+The Host header value is extracted from the hostname of the API's `upstream_url`
+field.
+
+However, by configuring your API with `preserve_host=true`:
```

Comment:
Should further clarify what adding this will do in the sentence, either before the examples in this sentence, or in the next sentence. It is currently relying on the example to imply it's action.